docs/reference/search/async-search.asciidoc

```diff
@@ -0,0 +1,210 @@
+      "hits" : [ ]
+    }
+  }
+}
+--------------------------------------------------
+// TESTRESPONSE[s/FmRldE8zREVEUzA2ZVpUeGs2ejJFUFEaMkZ5QTVrSTZSaVN3WlNFVmtlWHJsdzoxMDc=/$body.id/]
+// TESTRESPONSE[s/1583945890986/$body.start_time_in_millis/]
+// TESTRESPONSE[s/1584377890986/$body.expiration_time_in_millis/]
+// TESTRESPONSE[s/"took" : 1122/"took": $body.response.took/]
+// TESTRESPONSE[s/"total" : 562/"total": $body.response._shards.total/]
+// TESTRESPONSE[s/"successful" : 3/"successful": $body.response._shards.successful/]
+// TESTRESPONSE[s/"value" : 157483/"value": $body.response.hits.total.value/]
+
+<1> Identifier of the async search that can be used to monitor its progress, retrieve its results, and/or delete it.
+<2> Whether the returned search results are partial or final
+<3> Whether the search is still being executed or it has completed
+<4> How many shards the search will be executed on, overall
+<5> How many shards have successfully completed the search
+<6> How many documents are currently matching the query, which belong to the shards that have already completed the search
+
+It is possible to block and wait until the search is completed up to a certain
+timeout by providing the `wait_for_completion` parameter, which defaults to
+`1` second.
+
+You can also specify how long the async search needs to be
+available through the `keep_alive` parameter, which defaults to `5d` (five days).
+Ongoing async searches and any saved search results are deleted after this
+period.
+
+NOTE: When results are sorted by a numeric field, shards get sorted based on
+minimum and maximum value that they hold for that field, hence partial
+results become available following the sort criteria that was requested.
+
+The submit async search API supports the same <<search-search-api-query-params,parameters>>
+as the search API, though some have different default values:
+
+* `batched_reduce_size` defaults to `5`: this affects how often partial results
+become available., which happens whenever shard results are reduced. By default
+a partial reduction is performed every time the coordinating node has received
+5 new shard responses.
+* `request_cache` defaults to `true`
+* `pre_filter_shard_size` defaults to `1`: this is to enforce the execution of
+a pre-filter roundtrip to retrieve statistics from each shard so that the ones
+that surely don't hold any document matching the query can be skipped.
+* `ccs_minimize_roundtrips` defaults to `false`, which is also the only
+permitted value
+
+WARNING: Async search does not support <<request-body-search-scroll,scroll>>
+nor search requests that only include the  <<search-suggesters,suggest section>>.
+{ccs} is supported only with <<ccs-min-roundtrips,`ccs_minimize_roundtrips`>>
+set to `false`.
+
+[[get-async-search]]
+==== Get async search
+
+The get async search API retrieves the results of a previously submitted
+async search request given its id. The access to the results of a specific
```

Comment:
That's only when security is enabled that we restrict the access. We should make the distinction here.